libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -429,11 +429,13 @@ - (void)logoutUser:(SFUserAccount *)user
         return;
     }
 
-    if (user.isUserLoggingOut) {
-        [self log:SFLogLevelInfo msg:@"logoutUser: user is already in the process of logout."];
-        return;
+    @synchronized (user) {
```

Comment:
Kevin, Personally I would prefer if we used a single write-multi read pattern around the state, but I believe you are correct and this will address the crashing issue I am investigating. 

We can discuss extending exclusion to logging in, as well as documenting the expected behavior/use patterns  elsewhere.